src/reactors/PriorityOrderReactor.sol

```diff
@@ -56,17 +67,41 @@ contract PriorityOrderReactor is BaseReactor {
         );
     }
 
+    /// @notice update the priority order with the cosigner's data
+    /// @dev only called if the current block is before the auctionStartBlock signed by the user
+    /// @param orderHash the hash of the order
+    /// @param order the order to update
+    function _updateWithCosignerData(bytes32 orderHash, PriorityOrder memory order) internal pure {
+        /// return quickly if cosignerData is not set
+        if (order.cosignerData.auctionTargetBlock == 0) return;
+
+        if (order.cosignerData.auctionTargetBlock < order.auctionStartBlock) {
+            order.auctionStartBlock = order.cosignerData.auctionTargetBlock;
+        }
+        // validate cosigner signature
+        (bytes32 r, bytes32 s) = abi.decode(order.cosignature, (bytes32, bytes32));
+        uint8 v = uint8(order.cosignature[64]);
+        // cosigner signs over (orderHash || cosignerData)
+        address signer = ecrecover(keccak256(abi.encodePacked(orderHash, abi.encode(order.cosignerData))), v, r, s);
+        if (order.cosigner != signer || signer == address(0)) {
+            revert InvalidCosignature();
```

Comment:
What about if the user didn't specify a cosigner at all? Then this fails -- think we should allow that case right? Maybe w/ an early return above

Reply:
ah i thought that case would be caught by checking if the cosignerData auctionTargetBlock was the default val of 0

Reply:
oh, good point. Still adds a weird edge case where user didn't set a cosigner, but the filler sets a nonzero auctionTargetBloc and causes unnecessary failing

Reply:
yea agree - changing it to check cosigner == address(0), more intuitive